csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.cs

```diff
@@ -218,12 +218,28 @@ out nativeTensor
                                                     ));
 
                     // fill the native tensor, using GetValue(index) from the Tensor<string>
-                    string[] stringsInTensor = new string[tensorValue.Length];
-                    for (int i = 0; i < tensorValue.Length; i++)
+                    var len = tensorValue.Length;
+                    var stringsInTensor = new IntPtr[len];
+                    var pinnedHandles = new GCHandle[len + 1];
+                    pinnedHandles[len] = GCHandle.Alloc(stringsInTensor, GCHandleType.Pinned);
+                    try
                     {
-                        stringsInTensor[i] = tensorValue.GetValue(i);
+                        for (int i = 0; i < tensorValue.Length; i++)
+                        {
+                            var utf8str = UTF8Encoding.UTF8.GetBytes(tensorValue.GetValue(i) + "\0");
+                            pinnedHandles[i] = GCHandle.Alloc(utf8str, GCHandleType.Pinned);
+                            stringsInTensor[i] = pinnedHandles[i].AddrOfPinnedObject();
+                        }
+
+                        NativeApiStatus.VerifySuccess(NativeMethods.OrtFillStringTensor(nativeTensor, stringsInTensor, (UIntPtr)len));
```

Comment:
Should any exception be caught and logged ?

Reply:
It is caught below

---
In reply to: [360191590](https://github.com/microsoft/onnxruntime/pull/2697#discussion_r360191590) [](ancestors = 360191590)

Reply:
It goes into finally whether there is an exception or not. My question was more along the lines of - is a catch necessary before the finally (to log any abnormal occurrence) ?

Reply:
Missed the upper level catch. Got it now.